src/microbe_stage/Microbe.cs

```diff
@@ -1391,11 +1423,24 @@ internal void OnColonyMemberRemoved(Microbe microbe)
             RemoveCollisionExceptionWith(microbe);
     }
 
+    internal void ReParentShapes(Microbe to, Vector3 offset)
+    {
+        foreach (var organelle in organelles)
+            organelle.ReParentShapes(to, offset);
+    }
+
     internal void OnColonyMemberAdded(Microbe microbe)
     {
         if (microbe == this)
         {
             OnIGotAddedToColony();
+
+            if (Colony.Master != this)
+                Mode = ModeEnum.Static;
+
+            ReParentShapes(Colony.Master, (GlobalTransform.origin - Colony.Master.GlobalTransform.origin).Rotated(
```

Comment:
Form what i understood no,because it only gets the offset of the microbe that gets to be attached to the colony.

Reply:
But won't the positions of the organelle collisions or something need rotating to be correct?

Reply:
I dont see why you would want to rotate the shapes.You just move them according to teh offset right?

Reply:
But you need to first rotate them around the center of the child cell, then add the offset to the parent cell, right?

Reply:
So you need to rotate the child cell?Why?

Reply:
Because otherwise the visuals and the physics won't match. The position of the physics spheres changes based on the cell rotation, so that same rotation needs to be taken into account when adding the physics spheres to the colony leader, otherwise the spheres will attach as-if the colony child cell attached without being rotated at all (which I think is a very unlikely case).
I tried to draw something:
![trying_to_explain_stuff](https://user-images.githubusercontent.com/3796411/129394401-9cbdbfa5-ae9f-440d-b03d-9a5697874884.png)

Reply:
Oh i didnt know the microbe is rotated when added to colony yea good ,point thats why the shapes where sometimes out of memebrane bounds for childre.

Reply:
But couldnt i just get the rotation of this in the ReparentShapes method?Instead of passing along so many parameters?

Reply:
> But couldnt i just get the rotation of this in the ReparentShapes method?Instead of passing along so many parameters?

You could... but I already commented that I'd prefer to pass in the parameters to re-parent explicitly, instead of that method going back. So instead of code structure like this:
- Microbe does stuff
- Microbe calls reparent
- reparent in colony looks at the Microbe status data

it would be like this:
- Microbe does stuff
- Microbe precomputes (or fetches if it is just reading a property) data
- Microbe calls reparent, with all the "sensible" parameters.

I think that will improve the data flow logic (make this cleaner code and not spaghetti), when you have a clear direction of how the data advances to where it is needed.

Reply:
This requires fixing the colony rotation,with the other PR,if i rotatted the shapes correctly they would be still missaligned tho.

Reply:
@FuJa0815 made that rotation fix PR I linked above, which if it helps with the rotation, then it'll be correct. And then the physics positioning will need to be also updated to work.